### DISCUSSION TITLE
Compile process aborted on QNX real-time system

### DISCUSSION THREAD

**Discussion by NicolaBlasuttigh** [nlohmann/json#2374] [Q&A] [answered]
Hi everyone!
I'm trying to use json library on dSPACE SCALEXIO system (a QNX-based real-time hardware).

The first time I started the compilation, a lot of errors appeared immediately about the to_string() function that I fixed by following this [link](https://community.qnx.com/sf/discussion/do/listPosts/projects.toolchain/discussion.core_development_tools.topc26761?_pagenum=4) and then adding the relative `#include <custom/string.h>` in the json.hpp file.

Now a lot of other errors (which were probably already there before) such as:
- json.hpp:7130:28: error: 'strtoull' is not a member of 'std'.
             const auto x = std::strtoull(token_buffer.data(), &endptr, 10);
- same error with 'strtoll 
- same with 'snprintf
- other errors...
and a lot of warnings regarding uint8_t...

The compiler (ntox86-g++.exe (GCC) 5.2.0) writes me to include some alternatives like:
json.hpp:7146:28: note: suggested alternatives:
In file included: 
- from C:\...\QNX650_520\target\qnx6/usr/include/c++/5.2.0/cstdlib:34:0,
- from C:\...\QNX650_520\target\qnx6/usr/include/c++/5.2.0/bits/stl_algo.h:59,
- from C:\...\QNX650_520\target\qnx6/usr/include/c++/5.2.0/algorithm:62,...

Should I make some changes inside the json.hpp file by adding these libraries?
Is this the right solution?

Maybe the first question should be...can I use json library with Qnx?

If you want I can copy the entire error log but it is quite large.
...and if you need more information about the system, let me know!

Thank you.

**Comment by nlohmann** [comment, marked as answer]
Seems like the compiler has a lot of issues to compile C++11 code. As long as you can propose fixes that are within the standard, we could add them to the library.

**Reply by NicolaBlasuttigh** [reply]
I see! I would be happy to propose fixes to the library but I would have to figure out how to do it. The files suggested to me, are they files I should include? For example, strtoull is defined in stdlib.h as written [here](http://www.qnx.com/developers/docs/6.5.0/index.jsp?topic=%2Fcom.qnx.doc.neutrino_lib_ref%2Fs%2Fstrtoul.html) but even if I include the library in json.hpp it gives me the exact same error. 
Am I using the wrong method?
